Tests/Unit/Repository/PhpcrOdmRepositoryTest.php

```diff
@@ -138,4 +143,98 @@ public function testGetVersionsWillThrow()
 
         $this->getRepository()->getVersions('/test');
     }
+
+    /**
+     * @dataProvider provideAddInvalid
+     *
+     * @expectedException \InvalidArgumentException
+     */
+    public function testAddWillThrowForNonValidParameters($path, $resource, $noParent = false)
+    {
+        $this->documentManager->find(null, '/test')->willReturn($noParent ? null : $this->document);
+
+        $this->getRepository()->add($path, $resource);
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     */
+    public function testAddResourceNotMatchingParentPath()
+    {
+        $this->documentManager->find(null, '/test')->willReturn($this->document);
+        $this->resource->getPath()->willReturn('/bla/blub');
+
+        $this->getRepository()->add('/test', $this->resource);
+    }
+
+    public function testAddWillPersistResource()
+    {
+        $resource = new PhpcrOdmResource('/test', $this->document);
+
+        $this->documentManager->find(null, '/test')->willReturn($this->object);
+
+        $this->documentManager->persist($this->document)->shouldBeCalled();
+        $this->documentManager->flush()->shouldBeCalled();
+
+        $this->getRepository()->add('/test', $resource);
+    }
+
+    public function testAddWillPersistResourceCollection()
+    {
+        $resource = new PhpcrOdmResource('/test', $this->document);
+
+        $this->documentManager->find(null, '/test')->willReturn($this->object);
+
+        $this->documentManager->persist($this->document)->shouldBeCalled();
+        $this->documentManager->flush()->shouldBeCalled();
```

Comment:
Whats the diference except from the position in the test method (before/after the method call)?